<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>qseqeventframe</class>
 <widget class="QFrame" name="qseqeventframe">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>850</width>
    <height>417</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="MinimumExpanding" vsizetype="MinimumExpanding">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="maximumSize">
   <size>
    <width>900</width>
    <height>500</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Events</string>
  </property>
  <property name="frameShape">
   <enum>QFrame::NoFrame</enum>
  </property>
  <property name="frameShadow">
   <enum>QFrame::Plain</enum>
  </property>
  <property name="lineWidth">
   <number>3</number>
  </property>
  <property name="midLineWidth">
   <number>2</number>
  </property>
  <widget class="QTableWidget" name="eventTableWidget">
   <property name="geometry">
    <rect>
     <x>9</x>
     <y>17</y>
     <width>491</width>
     <height>380</height>
    </rect>
   </property>
   <property name="sizePolicy">
    <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <property name="minimumSize">
    <size>
     <width>440</width>
     <height>380</height>
    </size>
   </property>
   <property name="maximumSize">
    <size>
     <width>492</width>
     <height>999</height>
    </size>
   </property>
   <property name="toolTip">
    <string>Events in the pattern/sequence. Events can be edited only with the controls at the right.</string>
   </property>
   <property name="sizeAdjustPolicy">
    <enum>QAbstractScrollArea::AdjustToContents</enum>
   </property>
   <property name="alternatingRowColors">
    <bool>false</bool>
   </property>
   <property name="selectionBehavior">
    <enum>QAbstractItemView::SelectRows</enum>
   </property>
   <property name="gridStyle">
    <enum>Qt::DashLine</enum>
   </property>
   <property name="wordWrap">
    <bool>false</bool>
   </property>
   <property name="rowCount">
    <number>48</number>
   </property>
   <property name="columnCount">
    <number>7</number>
   </property>
   <attribute name="verticalHeaderMinimumSectionSize">
    <number>16</number>
   </attribute>
   <attribute name="verticalHeaderDefaultSectionSize">
    <number>18</number>
   </attribute>
   <row/>
   <row/>
   <row/>
   <row/>
   <row/>
   <row/>
   <row/>
   <row/>
   <row/>
   <row/>
   <row/>
   <row/>
   <row/>
   <row/>
   <row/>
   <row/>
   <row/>
   <row/>
   <row/>
   <row/>
   <row/>
   <row/>
   <row/>
   <row/>
   <row/>
   <row/>
   <row/>
   <row/>
   <row/>
   <row/>
   <row/>
   <row/>
   <row/>
   <row/>
   <row/>
   <row/>
   <row/>
   <row/>
   <row/>
   <row/>
   <row/>
   <row/>
   <row/>
   <row/>
   <row/>
   <row/>
   <row/>
   <row/>
   <column/>
   <column/>
   <column/>
   <column/>
   <column/>
   <column/>
   <column/>
  </widget>
  <widget class="QPlainTextEdit" name="plainTextEdit">
   <property name="geometry">
    <rect>
     <x>510</x>
     <y>270</y>
     <width>311</width>
     <height>69</height>
    </rect>
   </property>
   <property name="sizePolicy">
    <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <property name="lineWidth">
    <number>2</number>
   </property>
   <property name="horizontalScrollBarPolicy">
    <enum>Qt::ScrollBarAlwaysOff</enum>
   </property>
   <property name="placeholderText">
    <string>(Enter Meta/SysEx event text here)</string>
   </property>
  </widget>
  <widget class="QWidget" name="gridLayoutWidget">
   <property name="geometry">
    <rect>
     <x>520</x>
     <y>348</y>
     <width>261</width>
     <height>52</height>
    </rect>
   </property>
   <layout class="QGridLayout" name="gridLayout_2">
    <property name="sizeConstraint">
     <enum>QLayout::SetFixedSize</enum>
    </property>
    <property name="horizontalSpacing">
     <number>20</number>
    </property>
    <property name="verticalSpacing">
     <number>2</number>
    </property>
    <item row="0" column="2">
     <widget class="QPushButton" name="button_modify">
      <property name="enabled">
       <bool>false</bool>
      </property>
      <property name="minimumSize">
       <size>
        <width>64</width>
        <height>24</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>90</width>
        <height>24</height>
       </size>
      </property>
      <property name="toolTip">
       <string>Deletes the current event, then inserts a new event at the time shown above, with the data items shown above.</string>
      </property>
      <property name="text">
       <string>Modif&amp;y</string>
      </property>
     </widget>
    </item>
    <item row="0" column="0">
     <widget class="QPushButton" name="button_del">
      <property name="enabled">
       <bool>false</bool>
      </property>
      <property name="minimumSize">
       <size>
        <width>64</width>
        <height>24</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>90</width>
        <height>24</height>
       </size>
      </property>
      <property name="toolTip">
       <string>Deletes the currently-selected MIDI event.</string>
      </property>
      <property name="text">
       <string>&amp;Delete</string>
      </property>
     </widget>
    </item>
    <item row="1" column="2">
     <widget class="QPushButton" name="button_dump">
      <property name="enabled">
       <bool>false</bool>
      </property>
      <property name="minimumSize">
       <size>
        <width>64</width>
        <height>24</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>90</width>
        <height>24</height>
       </size>
      </property>
      <property name="toolTip">
       <string>Appends the text of events to a file for trouble-shooting.
It is in the same directory as the MIDI file, and has a name
of the form &quot;filename-pattern-#.text&quot;, where '#' is the number of
the pattern.</string>
      </property>
      <property name="text">
       <string>D&amp;ump</string>
      </property>
     </widget>
    </item>
    <item row="1" column="1">
     <widget class="QPushButton" name="button_save">
      <property name="enabled">
       <bool>false</bool>
      </property>
      <property name="minimumSize">
       <size>
        <width>64</width>
        <height>24</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>90</width>
        <height>24</height>
       </size>
      </property>
      <property name="toolTip">
       <string>Saves current changes to the pattern.  This step
must be done to preserve the changes.  Then the file must
be saved to make the changes permanent.  Tricky, but safer.</string>
      </property>
      <property name="text">
       <string>Store</string>
      </property>
     </widget>
    </item>
    <item row="0" column="1">
     <widget class="QPushButton" name="button_ins">
      <property name="enabled">
       <bool>false</bool>
      </property>
      <property name="minimumSize">
       <size>
        <width>64</width>
        <height>24</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>90</width>
        <height>24</height>
       </size>
      </property>
      <property name="toolTip">
       <string>Inserts a new event at the time shown above, with the data items shown above.</string>
      </property>
      <property name="text">
       <string>&amp;Insert</string>
      </property>
     </widget>
    </item>
    <item row="1" column="0">
     <widget class="QPushButton" name="button_clear">
      <property name="enabled">
       <bool>false</bool>
      </property>
      <property name="minimumSize">
       <size>
        <width>64</width>
        <height>24</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>90</width>
        <height>24</height>
       </size>
      </property>
      <property name="toolTip">
       <string>Deletes all of the events in this pattern.</string>
      </property>
      <property name="text">
       <string>Clear</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QWidget" name="vertLayoutWidget">
   <property name="geometry">
    <rect>
     <x>510</x>
     <y>107</y>
     <width>321</width>
     <height>158</height>
    </rect>
   </property>
   <layout class="QVBoxLayout" name="verticalLayout_2">
    <item>
     <layout class="QGridLayout" name="gridLayout" columnstretch="1,0">
      <property name="sizeConstraint">
       <enum>QLayout::SetFixedSize</enum>
      </property>
      <property name="spacing">
       <number>2</number>
      </property>
      <item row="1" column="0">
       <widget class="QLabel" name="time_label">
        <property name="minimumSize">
         <size>
          <width>72</width>
          <height>24</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>72</width>
          <height>24</height>
         </size>
        </property>
        <property name="font">
         <font>
          <bold>true</bold>
         </font>
        </property>
        <property name="text">
         <string>Time  </string>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="QComboBox" name="channel_combo_box">
        <property name="minimumSize">
         <size>
          <width>208</width>
          <height>28</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>224</width>
          <height>28</height>
         </size>
        </property>
        <property name="editable">
         <bool>true</bool>
        </property>
        <property name="currentIndex">
         <number>-1</number>
        </property>
        <property name="maxVisibleItems">
         <number>12</number>
        </property>
        <property name="maxCount">
         <number>17</number>
        </property>
        <property name="insertPolicy">
         <enum>QComboBox::NoInsert</enum>
        </property>
        <property name="minimumContentsLength">
         <number>1</number>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QComboBox" name="combo_ev_name">
        <property name="minimumSize">
         <size>
          <width>208</width>
          <height>28</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>224</width>
          <height>28</height>
         </size>
        </property>
        <property name="editable">
         <bool>true</bool>
        </property>
        <property name="maxVisibleItems">
         <number>12</number>
        </property>
        <property name="maxCount">
         <number>32</number>
        </property>
        <property name="insertPolicy">
         <enum>QComboBox::NoInsert</enum>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QLineEdit" name="entry_ev_timestamp">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>208</width>
          <height>28</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>224</width>
          <height>28</height>
         </size>
        </property>
        <property name="toolTip">
         <string>Current time in bars:beats:pulses. Edit to insert or modify events.</string>
        </property>
        <property name="text">
         <string>001:1:000</string>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QLabel" name="channel_label">
        <property name="minimumSize">
         <size>
          <width>72</width>
          <height>24</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>72</width>
          <height>24</height>
         </size>
        </property>
        <property name="font">
         <font>
          <bold>true</bold>
         </font>
        </property>
        <property name="text">
         <string>Chan</string>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="event_label">
        <property name="minimumSize">
         <size>
          <width>72</width>
          <height>24</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>72</width>
          <height>24</height>
         </size>
        </property>
        <property name="font">
         <font>
          <bold>true</bold>
         </font>
        </property>
        <property name="text">
         <string>Event</string>
        </property>
       </widget>
      </item>
      <item row="0" column="0">
       <widget class="QLabel" name="label_category">
        <property name="minimumSize">
         <size>
          <width>72</width>
          <height>24</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>72</width>
          <height>24</height>
         </size>
        </property>
        <property name="font">
         <font>
          <bold>true</bold>
         </font>
        </property>
        <property name="text">
         <string>Category</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QComboBox" name="combo_ev_category">
        <property name="minimumSize">
         <size>
          <width>208</width>
          <height>24</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>224</width>
          <height>24</height>
         </size>
        </property>
        <property name="maxVisibleItems">
         <number>8</number>
        </property>
        <property name="maxCount">
         <number>8</number>
        </property>
        <property name="insertPolicy">
         <enum>QComboBox::NoInsert</enum>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QGridLayout" name="gridLayout_3" columnstretch="0,0,0,0,0,0">
      <property name="sizeConstraint">
       <enum>QLayout::SetFixedSize</enum>
      </property>
      <property name="horizontalSpacing">
       <number>8</number>
      </property>
      <item row="0" column="3">
       <widget class="QLineEdit" name="entry_ev_data_1">
        <property name="minimumSize">
         <size>
          <width>36</width>
          <height>24</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>40</width>
          <height>24</height>
         </size>
        </property>
        <property name="statusTip">
         <string>The second data byte, if applicable.  Change using decimal or hex (0xnn) notation.</string>
        </property>
        <property name="text">
         <string>0x00</string>
        </property>
       </widget>
      </item>
      <item row="0" column="2">
       <widget class="QLabel" name="d1_label">
        <property name="minimumSize">
         <size>
          <width>24</width>
          <height>24</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>24</width>
          <height>24</height>
         </size>
        </property>
        <property name="font">
         <font>
          <bold>true</bold>
         </font>
        </property>
        <property name="text">
         <string>D1</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
        </property>
       </widget>
      </item>
      <item row="0" column="4">
       <widget class="QCheckBox" name="pulse_time_check_box">
        <property name="minimumSize">
         <size>
          <width>64</width>
          <height>24</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>64</width>
          <height>24</height>
         </size>
        </property>
        <property name="text">
         <string>Ticks</string>
        </property>
       </widget>
      </item>
      <item row="0" column="5">
       <widget class="QCheckBox" name="hex_data_check_box">
        <property name="text">
         <string>Hex</string>
        </property>
       </widget>
      </item>
      <item row="0" column="0">
       <widget class="QLabel" name="d0_label">
        <property name="minimumSize">
         <size>
          <width>24</width>
          <height>24</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>24</width>
          <height>24</height>
         </size>
        </property>
        <property name="font">
         <font>
          <bold>true</bold>
         </font>
        </property>
        <property name="text">
         <string>D0</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QLineEdit" name="entry_ev_data_0">
        <property name="minimumSize">
         <size>
          <width>36</width>
          <height>24</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>40</width>
          <height>24</height>
         </size>
        </property>
        <property name="statusTip">
         <string>The first data byte, if applicable. Change using decimal or hex (0xnn) notation.</string>
        </property>
        <property name="text">
         <string>0x00</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QWidget" name="horizLayoutWidget">
   <property name="geometry">
    <rect>
     <x>510</x>
     <y>12</y>
     <width>321</width>
     <height>32</height>
    </rect>
   </property>
   <layout class="QHBoxLayout" name="horizontalLayout_2" stretch="0,0">
    <property name="spacing">
     <number>6</number>
    </property>
    <property name="sizeConstraint">
     <enum>QLayout::SetMinimumSize</enum>
    </property>
    <property name="rightMargin">
     <number>10</number>
    </property>
    <item>
     <widget class="QLabel" name="label_seq_number">
      <property name="minimumSize">
       <size>
        <width>36</width>
        <height>0</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>48</width>
        <height>16777215</height>
       </size>
      </property>
      <property name="font">
       <font>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>1024</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QLineEdit" name="m_entry_name">
      <property name="sizePolicy">
       <sizepolicy hsizetype="MinimumExpanding" vsizetype="Fixed">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>248</width>
        <height>24</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>256</width>
        <height>24</height>
       </size>
      </property>
      <property name="font">
       <font>
        <family>Anonymous Pro</family>
        <pointsize>12</pointsize>
       </font>
      </property>
      <property name="text">
       <string>Loop Name</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QWidget" name="dataLayoutWidget">
   <property name="geometry">
    <rect>
     <x>510</x>
     <y>48</y>
     <width>321</width>
     <height>58</height>
    </rect>
   </property>
   <layout class="QGridLayout" name="gridLayout_4" rowstretch="0,0" columnstretch="1,6">
    <property name="sizeConstraint">
     <enum>QLayout::SetFixedSize</enum>
    </property>
    <property name="verticalSpacing">
     <number>2</number>
    </property>
    <item row="0" column="0">
     <widget class="QPushButton" name="button_link">
      <property name="minimumSize">
       <size>
        <width>64</width>
        <height>24</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>64</width>
        <height>24</height>
       </size>
      </property>
      <property name="font">
       <font>
        <pointsize>8</pointsize>
        <italic>true</italic>
       </font>
      </property>
      <property name="toolTip">
       <string>If enabled, this setting allows two linked
Note events to be selected at the
same time.</string>
      </property>
      <property name="text">
       <string>Sel Linked</string>
      </property>
      <property name="checkable">
       <bool>true</bool>
      </property>
      <property name="flat">
       <bool>false</bool>
      </property>
     </widget>
    </item>
    <item row="0" column="1">
     <widget class="QLabel" name="label_time_sig">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>248</width>
        <height>24</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>248</width>
        <height>24</height>
       </size>
      </property>
      <property name="text">
       <string>Time Signature and PPQN</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
    </item>
    <item row="1" column="0">
     <widget class="QPushButton" name="button_blank">
      <property name="minimumSize">
       <size>
        <width>64</width>
        <height>24</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>64</width>
        <height>16777215</height>
       </size>
      </property>
      <property name="text">
       <string/>
      </property>
      <property name="flat">
       <bool>true</bool>
      </property>
     </widget>
    </item>
    <item row="1" column="1">
     <widget class="QLabel" name="label_measures_ev_count">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>248</width>
        <height>24</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>248</width>
        <height>24</height>
       </size>
      </property>
      <property name="text">
       <string>Measures and Events Count</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <tabstops>
  <tabstop>eventTableWidget</tabstop>
  <tabstop>entry_ev_timestamp</tabstop>
  <tabstop>combo_ev_name</tabstop>
  <tabstop>channel_combo_box</tabstop>
  <tabstop>entry_ev_data_0</tabstop>
  <tabstop>entry_ev_data_1</tabstop>
  <tabstop>button_del</tabstop>
  <tabstop>button_clear</tabstop>
  <tabstop>button_ins</tabstop>
  <tabstop>button_save</tabstop>
  <tabstop>button_modify</tabstop>
  <tabstop>button_dump</tabstop>
 </tabstops>
 <resources/>
 <connections/>
</ui>
